docs/reference/settings/data-stream-lifecycle-settings.asciidoc

```diff
@@ -10,6 +10,18 @@ These are the settings available for configuring <<data-stream-lifecycle, data s
 
 ==== Cluster level settings
 
+[[data-streams-lifecycle-retention-max]]
+`data_streams.lifecycle.retention.max`::
+(<<dynamic-cluster-setting,Dynamic>>, <<time-units, time unit value>>)
+The maximum retention period that will apply to all user data streams managed by the data stream lifecycle. The max retention will also
+override the retention of a data stream whose configured retention exceeds the max retention. It should be greater than `10s`.
```

Comment:
Do we want to give the defaults for these (unlimited)? Or are you intentionally leaving that out since the default will be different in different deployments?

Reply:
Exactly. We do not want (or even should not) implicitly delete users' data. However, solutions do have the opportunity to set it to a value that makes sense for their use case.